pkg/scheduler/scheduler.go

```diff
@@ -183,6 +194,10 @@ func (s *Scheduler) schedule(ctx context.Context) wait.SpeedSignal {
 	log := ctrl.LoggerFrom(ctx).WithValues("schedulingCycle", s.schedulingCycle)
 	ctx = ctrl.LoggerInto(ctx, log)
 
+	if features.Enabled(features.AdmissionFairSharing) {
```

Comment:
Please also add a condition that this is executed when the penalty map is not empty